src/parser/jobs/mnk/modules/RiddleOfFire.tsx

```diff
@@ -24,69 +26,73 @@ const ROF_GCD = {
 	ERROR: 7,
 }
 
+class Riddle {
+	casts: CastEvent[]
+	start: number
+	end?: number | null = null
+	rushing: boolean = false
+
+	constructor(start: number) {
+		this.start = start
+		this.casts = []
+	}
+}
+
 export default class RiddleOfFire extends Module {
 	static handle = 'riddleoffire'
-	static dependencies = [
-		'suggestions',
-	]
-
 	static title = t('mnk.rof.title')`Riddle of Fire`
 	static displayOrder = DISPLAY_ORDER.RIDDLE_OF_FIRE
 
-	_active = false
-	_history = []
-	_riddle = {casts: []}
-	_rushing = false
+	@dependency private suggestions!: Suggestions
 
-	_missedGcds = 0
-	_missedTks = 0
+	_active: boolean = false
+	_history: Riddle[] = []
+	_riddle!: Riddle
+	_rushing: boolean = false
 
-	constructor(...args) {
-		super(...args)
-		this.addHook('cast', {by: 'player'}, this._onCast)
-		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.RIDDLE_OF_FIRE.id}, this._onDrop)
-		this.addHook('complete', this._onComplete)
+	_missedGcds: number = 0
+	_missedTks: number = 0
+
+	protected init(): void {
+		this.addHook('cast', {by: 'player'}, this.onCast)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.RIDDLE_OF_FIRE.id}, this.onDrop)
+		this.addHook('complete', this.onComplete)
 	}
 
-	_onCast(event) {
-		const actionId = event.ability.guid
+	onCast(event: CastEvent): void {
+		const action = getDataBy(ACTIONS, 'id', event.ability.guid) as TODO // should be Action type
 
-		if (actionId === ACTIONS.RIDDLE_OF_FIRE.id) {
+		if (action.id === ACTIONS.RIDDLE_OF_FIRE.id) {
```

Comment:
Technically speaking, `getDataBy<T>` has a return type of `T | undefined`, so you should be existence checking it.